server/src/FileChangeListener.ts

```diff
@@ -38,11 +38,12 @@ export class FileChangeListener {
   }
 
   updateDbtProjectConfig(): void {
-    this.dbtRepository.dbtTargetPath = this.dbtProject.findTargetPath();
-    this.dbtRepository.projectName = this.dbtProject.findProjectName();
-    this.dbtRepository.macroPaths = this.dbtProject.findMacroPaths();
-    this.dbtRepository.modelPaths = this.dbtProject.findModelPaths();
-    this.dbtRepository.packagesInstallPaths = this.dbtProject.findPackagesInstallPaths();
+    const dbtProject = this.dbtProject.getProject();
```

Comment:
If I understand it correctly, we can mark dbtProject as `updated` when it is actually updated. And use the following logic (instead of passing argument and control update state outside the class):
```
 getProject(): Record<string, unknown> {
   if(this.updated || !this.parsedProject) {
      this.parsedProject = YamlParserUtils.parseYamlFile(path.resolve(this.projectPath, DbtRepository.DBT_PROJECT_FILE_NAME)) as Record<string, unknown>;
    }
    return this.parsedProject;
  }
```

Reply:
Good point. Parse results caching implemented.